stacks/spica/src/passport/services/apikey.service.ts

```diff
@@ -0,0 +1,69 @@
+import {Injectable} from "@angular/core";
```

Comment:
This is the wrong way to mock services in the hierarchical injector. We don't and can't use interfaces to make abstractions cause they are tripped away after the compilation.

Reply:
Refer to this article: https://blog.danieleghidoli.it/2016/11/06/testing-angular-component-mock-services/

Reply:
The article says there are two ways to mock service, mocking whole class or mocking only methods that will use on test. If you asked why I mocked the whole class, I was developing with mock class before backend service ready to use so that's why I prefer mocking class instead of mocking methods. On the other hand, I couldn't understand the relation between your first and second comments :)

Reply:
If you want to mock something, first you create an abstract class to define the methods and structure. After that, you create a mock class that extends that class and you provide this mock class to the hierarchical injector.

Reply:
Example;

```

abstract class Test {
  abstract do(): string;
}

class MockTest  extends Test{
  do(): string {
   return "MockTest.do";
  }
}


// Module
{
  provide: Test,
  useValue: new MockTest()
}

// Component
class AComp{
  constructor(testService: Test) {
    console.log(testService.do()) // Will print MockTest.do to the console
  }
}
```